torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -820,8 +828,13 @@ void Reducer::initialize_buckets(
           variable_index < variable_locators_.size(),
           "Out of range variable index specified.");
       variable_locators_[variable_index] = VariableLocator{
+#ifdef _WIN32
+          bucket_index,
+          intra_bucket_index++,
+#else
           .bucket_index = bucket_index,
           .intra_bucket_index = intra_bucket_index++,
+#endif
```

Comment:
This looks a bit ugly, perhaps we can just use a non-name-prefixed initialization on all platforms?